app/src/main/java/com/cornellappdev/android/eateryblue/ui/viewmodels/state/EateryRetrievalState.kt

```diff
@@ -1,10 +1,25 @@
 package com.cornellappdev.android.eateryblue.ui.viewmodels.state
 
+import com.cornellappdev.android.eateryblue.ui.viewmodels.state.EateryApiResponse.Error
+import com.cornellappdev.android.eateryblue.ui.viewmodels.state.EateryApiResponse.Pending
+import com.cornellappdev.android.eateryblue.ui.viewmodels.state.EateryApiResponse.Success
+
+// TODO: Delete this, and replace it with [EateryApiResponse]. Also, replace all occurrences of this
+//  with a Flow-based networking model. See EateryDetailViewModel for an example of using flow
+//  mapping / combination to best fit with Jetpack Compose.
+sealed class EateryRetrievalState {
+    object Pending : EateryRetrievalState()
+    object Error : EateryRetrievalState()
+    object Success : EateryRetrievalState()
+}
+
 /**
- * A sealed hierarchy describing the current status of retrieving any eateries.
+ * Represents the state of an api response fetching data of type [T].
+ * Can be: [Pending], which represents the call still loading in, [Error], which represents the
+ * API call failing, and [Success], which contains a `data` field containing the [T] data.
  */
-sealed interface EateryRetrievalState {
-    object Success : EateryRetrievalState
-    object Error : EateryRetrievalState
-    object Pending : EateryRetrievalState
+sealed class EateryApiResponse<out T : Any> {
+    object Pending : EateryApiResponse<Nothing>()
+    object Error : EateryApiResponse<Nothing>()
+    class Success<out T : Any>(val data: T) : EateryApiResponse<T>()
```

Comment:
Love to see the generic response state class! Volume really needs this, we have separate response states for every object in our code base for some reason.

Reply:
I used this same structure in Uplift, and although it makes a bit of headache with pattern matching the Pending, Error, Success everytime an API call is needed, it makes for a really nice and streamlined structure!